app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt

```diff
@@ -0,0 +1,23 @@
+package com.thefuntasty.mvvmsample.domain
+
+import com.thefuntasty.interactors.BaseMayber
+import io.reactivex.Maybe
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class StateInteractor @Inject constructor() : BaseMayber<Boolean>() {
+
+    private var emitSuccess: Boolean = false
+
+    fun init(emitSuccess: Boolean) = apply {
+        this.emitSuccess = emitSuccess
+    }
+
+    override fun prepare(): Maybe<Boolean> = if (emitSuccess) {
+        Maybe.fromCallable {
+            true
+        }.delay(5, TimeUnit.SECONDS)
+    } else {
+        Maybe.empty<Boolean>().delay(5, TimeUnit.SECONDS)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt#L21"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>